src/Stack.spec.ts

```diff
@@ -0,0 +1,70 @@
+import {IRequest, IRequestHeaders} from "./Client";
+import {IMiddleware, Stack} from "./Stack";
+
+describe("Stack", () => {
+    it("should be defined", () => {
+        const stack = new Stack();
+        expect(stack).toBeDefined();
+    });
+    it("Should accept function which returns defaultOptions", () => {
+        const def = () => {
+            const headers: IRequestHeaders = {
+                "Accept": "",
+                "Authorization": "",
+                "Content-Type": "application/json",
+                "Accept-Language": "en"
+            };
+            return {
+                baseUrl: "http://api.example.com/example",
+                body: JSON.stringify({user: 1}),
+                method: "POST",
+                queryParameters: {user: "1", id: "2"},
+                url: "/products",
+                headers
+            } as IRequest;
+        };
+        const stack = new Stack(def);
+        expect(stack.defaultOptions).toBeDefined();
```

Comment:
same as Stack... this is not actually testing if the functionality provided is working properly.